integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -206,6 +208,30 @@ case class CarbonIUDAnalysisRule(sparkSession: SparkSession) extends Rule[Logica
           table)
     }
   }
+
+
+  /**
+   * This method will check whether the data type of new value is supported or not
+   * For example, arrayType is not supported
+   */
+  def checkForUnsupportedType(selectStmt: String): Unit = {
+    // selectStmt will have "select 'newValue' from table"
+    // we can validate the new value by executing the select query
+    val sel = selectStmt.substring(0, selectStmt.lastIndexOf("from")).trim;
```

Comment:
Better not to add this type of check , 
replace dataSet = sparkSession.sql(selectStmt).collect() directly for any case it will execute and get the rows